repose-aggregator/commons/commons-utilities/src/main/scala/org/openrepose/commons/utils/opentracing/httpclient/ReposeTracingRequestInterceptor.scala

```diff
@@ -19,47 +19,77 @@
+  * [[com.uber.jaeger.httpclient.SpanInjectionRequestInterceptor]] were collapsed inline.
   *
   * @param tracer a [[io.opentracing.Tracer]] to bridge this utility with the OpenTracing API
   */
-class ReposeTracingRequestInterceptor(tracer: Tracer, reposeVersion: String, uriRedactionService: UriRedactionService)
-  extends SpanCreationRequestInterceptor(tracer) {
-
-  private val spanInjectionInterceptor = new SpanInjectionRequestInterceptor(tracer)
+class ReposeTracingRequestInterceptor(tracer: Tracer, reposeVersion: String, uriRedactionService: UriRedactionService) extends HttpRequestInterceptor with StrictLogging {
 
+  @throws[HttpException]
+  @throws[IOException]
   override def process(httpRequest: HttpRequest, httpContext: HttpContext): Unit = {
-    super.process(httpRequest, httpContext)
+    try {
+      val currentActiveSpan = tracer.activeSpan
+      val requestLine = httpRequest.getRequestLine
+      val redactedUri = uriRedactionService.redact(requestLine.getUri)
 
-    spanInjectionInterceptor.process(httpRequest, httpContext)
-  }
+      val clientSpanBuilder = tracer.buildSpan(s"${requestLine.getMethod} $redactedUri")
+      if (currentActiveSpan != null) clientSpanBuilder.asChildOf(currentActiveSpan)
 
-  override protected def onSpanStarted(clientSpan: Span, httpRequest: HttpRequest, httpContext: HttpContext): Unit = {
-    Option(httpRequest.getFirstHeader(REQUEST_ID))
-      .map(_.getValue)
-      .foreach(clientSpan.setTag(REQUEST_ID, _))
-    Option(httpRequest.getFirstHeader(VIA))
-      .map(_.getValue)
-      .foreach(clientSpan.setTag(VIA, _))
-    clientSpan.setTag(ReposeVersion, reposeVersion)
-    // Replace the http.url tag so that we do not leak the raw URL being set in this tag by the underlying Jaeger interceptor implementation
-    clientSpan.setTag(HTTP_URL.toString, uriRedactionService.redact(httpRequest.getRequestLine.getUri))
-  }
+      val clientSpan = clientSpanBuilder.start
+
+      clientSpan.setTag(SPAN_KIND.toString, SPAN_KIND_CLIENT)
+      clientSpan.setTag(HTTP_URL.toString, redactedUri)
+
+      httpContext match {
+        case context: HttpClientContext =>
+          val host = context.getTargetHost
+          clientSpan.setTag(PEER_HOSTNAME.toString, host.getHostName)
+          clientSpan.setTag(PEER_PORT.toString, host.getPort)
+        case _ =>
+      }
 
-  override protected def getOperationName(httpRequest: HttpRequest): String =
-    s"${httpRequest.getRequestLine.getMethod} ${uriRedactionService.redact(httpRequest.getRequestLine.getUri)}"
+      Option(httpRequest.getFirstHeader(REQUEST_ID))
+        .map(_.getValue)
+        .foreach(clientSpan.setTag(REQUEST_ID, _))
+      Option(httpRequest.getFirstHeader(VIA))
+        .map(_.getValue)
+        .foreach(clientSpan.setTag(VIA, _))
+      clientSpan.setTag(ReposeVersion, reposeVersion)
+
+      httpContext.setAttribute("io.opentracing.Span", clientSpan)
```

Comment:
Extract this to a constant somewhere so that it is always correct between the request and response interceptors?

Reply:
Yep. I can't believe I missed that one. Updated.